src/Lessons-api.Data/Repositories/TeacherRepository.cs

```diff
@@ -39,28 +40,31 @@ public async Task<TeacherEntity> AddTeacher(TeacherEntity teacherEntity)
             return teacherEntity;
         }
 
-        public async Task<TeacherEntity> UpdateTeacher(int id, TeacherEntity teacherEntity)
+        public async Task<TeacherEntity> UpdateTeacher(int id, UserEntity userEntity)
         {
-            var updatedTeacher = await _lessonsContext.Teachers.FindAsync(id);
+            var updatedTeacher = await _lessonsContext.Teachers.Where(t => t.Id == id).Include(t => t.User).FirstOrDefaultAsync();
```

Comment:
Use FindAsync

Reply:
When I use FindAsync () I have some fields null, I did this for include ().